src/main/java/mate/academy/bookshop/specification/book/impl/TitleSpecification.java

```diff
@@ -0,0 +1,24 @@
+package mate.academy.bookshop.specification.book.impl;
+
+import java.util.Arrays;
+import mate.academy.bookshop.dto.BookSearchParameters;
+import mate.academy.bookshop.model.Book;
+import mate.academy.bookshop.specification.book.BookSpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TitleSpecification implements BookSpecificationProvider {
+    private static final String COLUMN_NAME = "title";
+
+    @Override
+    public Specification<Book> getSpecification(BookSearchParameters bookSearchParameters) {
+        return (root, query, criteriaBuilder)
+                -> root.get(COLUMN_NAME).in(Arrays.stream(bookSearchParameters.titles()).toArray());

```

Comment:
you see, it makes sense to filter some books by the author, but I do not see much sense in filtering books by their title, people usually search by the names of the entity, so let's do some refactors on that, and receive a single `searchString` param in the controller, which  we will then pass somewhere down here where we will make `SearchSpecificationProvider` which will be working by SQL `like` principle, for ex: `root.get(TITLE_KEY).like(searchString)` or maybe ignorecase like or something like that